[[ Bug 19045 ]] Fix surrogate processing in MCUnicodeGetProperty

runrevmark opened the pull request:
This patch fixes the surrogate processing in MCUnicodeGetProperty
which was previously computing the codepoint incorrectly.

runrevmark committed:
[[ Bug 19045 ]] Fix surrogate processing in MCUnicodeGetProperty

This patch fixes the surrogate processing in MCUnicodeGetProperty
which was previously computing the codepoint incorrectly.

livecodeali commented:
@livecode-vulcan review ok 437dc69

livecode-vulcan commented:
💙 review by @livecodeali ok 437dc69b5d132055dd0abcf92c97986c29bcf114

livecode-vulcan commented:
😞 test failure 437dc69b5d132055dd0abcf92c97986c29bcf114

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/763)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/768)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/761)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/762)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/768)
* try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/760)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/760)

runrevmark committed:
[[ Bug 19045 ]] Make sure test compiles on windows

This patch removes the use of a dynamic stack allocated array
which the version of MSVC we use does not support.

peter-b commented:
@livecode-vulcan review ok 5deb2cd

livecode-vulcan commented:
💙 review by @peter-b ok 5deb2cdb78f4091cdadfea6f05438d163c6a35b2

livecode-vulcan commented:
😎 test success 5deb2cdb78f4091cdadfea6f05438d163c6a35b2

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/764)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/769)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/762)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/763)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/769)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/761)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/761)
